File: 백준/Gold/1011. Fly me to the Alpha Centauri/Fly me to the Alpha Centauri.py
# Fly me to the Alpha Centauri
import sys
input = sys.stdin.readline

N = int(input())

for i in range(N):
    start, to = map(int,input().split())

    to = to - start
    start = 0
    
    if to - start <= 3:
        print(to - start)
        continue

    speed = int(to**0.5)
    rest = to - speed**2
    
    if rest % speed:
        print(2*speed + rest // speed)
    else:
        print(2*speed - 1 + rest // speed)

File: 백준/Gold/1011. Fly me to the Alpha Centauri/Fly me to the Alpha Centauri.java
import java.util.*;
import java.io.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;

        int N = Integer.parseInt(br.readLine());

        for (int i = 0; i < N; i++) {
            st = new StringTokenizer(br.readLine());

            int start = Integer.parseInt(st.nextToken());
            int to = Integer.parseInt(st.nextToken());

            to = to - start;
            start = 0;
            
            int speed = (int) Math.sqrt(to);
            int rest = to - speed * speed;

            if (rest % speed != 0) {
                System.out.println(2*speed + rest / speed);
            } else{
                System.out.println(2*speed -1 + rest / speed);
            }
        }
    }
}
